File: api/create-invoice.py
import json
import urllib.request
from http.server import BaseHTTPRequestHandler

BOT_TOKEN = "8872260684:AAHU65LnhHmLAItW3J6ECA-l9RyAOaSwAy8"

class handler(BaseHTTPRequestHandler):
    def do_POST(self):
        try:
            content_length = int(self.headers.get('Content-Length', 0))
            post_data = json.loads(self.rfile.read(content_length)) if content_length > 0 else {}
            stars = int(post_data.get('stars', 5))

            url = f"https://api.telegram.org/bot{BOT_TOKEN}/createInvoiceLink"
            payload = json.dumps({
                "title": "Попытка в рулетке",
                "description": f"Вращение за {stars} зв.",
                "payload": f"spin_{stars}_stars",
                "provider_token": "",
                "currency": "XTR",
                "prices": [{"label": "Звезды", "amount": stars}]
            }).encode('utf-8')

            req = urllib.request.Request(
                url, 
                data=payload, 
                headers={'Content-Type': 'application/json'}
            )
            
            with urllib.request.urlopen(req) as response:
                result = json.loads(response.read().decode('utf-8'))
                invoice_url = result.get('result', '')

                self.send_response(200)
                self.send_header('Content-Type', 'application/json')
                self.send_header('Access-Control-Allow-Origin', '*')
                self.send_header('Access-Control-Allow-Methods', 'POST, OPTIONS')
                self.send_header('Access-Control-Allow-Headers', 'Content-Type')
                self.end_headers()
                self.wfile.write(json.dumps({'invoice_url': invoice_url}).encode('utf-8'))

        except Exception as e:
            self.send_response(500)
            self.send_header('Content-Type', 'application/json')
            self.send_header('Access-Control-Allow-Origin', '*')
            self.end_headers()
            self.wfile.write(json.dumps({'error': str(e)}).encode('utf-8'))

    def do_OPTIONS(self):
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()

File: api/create-invoice.js
export default async function handler(req, res) {
    // Разрешаем кросс-доменные запросы (CORS)
    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Access-Control-Allow-Methods', 'POST, OPTIONS');
    res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

    if (req.method === 'OPTIONS') return res.status(200).end();
    if (req.method !== 'POST') return res.status(405).json({ error: 'Method Not Allowed' });

    // ТОКЕН ВАШЕГО БОТА
    const BOT_TOKEN = "8872260684:AAHU65LnhHmLAItW3J6ECA-l9RyAOaSwAy8";
    const { stars = 5 } = req.body || {};

    try {
        const response = await fetch(`https://api.telegram.org/bot${BOT_TOKEN}/createInvoiceLink`, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({
                title: "Попытка в рулетке",
                description: `Вращение рулетки за ${stars} звезд`,
                payload: `spin_${stars}_stars`,
                provider_token: "", // Для Telegram Stars оставляем пустым
                currency: "XTR",   // Код валюты Telegram Stars
                prices: [{ label: "Звезды", amount: Number(stars) }]
            })
        });

        const data = await response.json();

        if (data.ok) {
            // Возвращаем готовую ссылку t.me/$...
            return res.status(200).json({ invoice_url: data.result });
        } else {
            return res.status(400).json({ error: data.description || 'Ошибка Telegram API' });
        }
    } catch (error) {
        return res.status(500).json({ error: error.message });
    }
}
